### PR TITLE
MSPSDS-231: Image management functionality for investigations and products

### PR DESCRIPTION
**Pull request by tom-bell-softwire**

### CODE REVIEW

File: app/assets/stylesheets/images.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Place all the styles related to the Images controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+@import "buttons";
+@import "colours";
+
+.image-with-preview {
+  background-color: $grey-1;
```
Comment (broder) [inline app/assets/stylesheets/images.scss]: Are these styles common between the three image states? Could we move them into the ```.image-list {
  .image-with-preview,
  .image-without-preview,
  .image-processing-placeholder {``` bit below?

File: app/models/rapex_image.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class Image < ApplicationRecord
+class RapexImage < ApplicationRecord
```
Comment (broder) [inline app/models/rapex_image.rb]: Should we put a `TODO` to remove this and upload the RAPEX images into our image storage or do you think we should just keep hotlinking to RAPEX?

File: app/views/investigations/show.html.erb [hunk 3]
```diff
@@ -68,9 +68,9 @@
         <h2 class="govuk-heading-m">
           Images
         </h2>
-        <% if product.images.any? %>
```
[the remark this answers is not in the document]
Comment (broder) [reply app/views/investigations/show.html.erb]: Yeah, I think so - might be related to the comment about removing `RapexImage`s above as well

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 # rubocop:disable Metrics/BlockLength
 Rails.application.routes.draw do
-  resources :investigations do
+  concern :image_attachable do
+    resources :images
```
Comment (broder) [inline config/routes.rb]: Neat!

File: app/models/rapex_image.rb (hunk 2 above)
Comment (tom-bell-softwire) [reply app/models/rapex_image.rb]: Good call – I'll add a TODO (and related JIRA ticket).

I'd prefer to replace this with using stored images across the board. It just makes the logic in the views simpler.

File: app/views/investigations/show.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (tom-bell-softwire) [reply app/views/investigations/show.html.erb]: Cool. I'd add a TODO.

File: app/assets/stylesheets/images.scss (hunk 1 above)
Comment (tom-bell-softwire) [reply app/assets/stylesheets/images.scss]: These are used in the `_image` and `_image_tag` partials, which are used in several places (investigation and product pages, plus the various manage image pages).

They're then extended within `.image-list` and `investigations.scss` to specify the thumbnail size for the relevant page. However, there might be a neater way to achieve that.

I think I need to tweak some of the styling slightly anyway, so will have another look as part of that.

**Pull request merged**